commit 709ae4bc59edc7ece4cae4b899ba4aaa6385cd44 (refs/changes/69/4469/1)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2018-07-02T17:26:13-07:00 (1 year, 3 months ago)
    
    MANTA-3807 waferlock should have nicer errors when VMAPI reqs time out

diff --git a/lib/sapi.js b/lib/sapi.js
index 14675a3..2f739b1 100644
--- a/lib/sapi.js
+++ b/lib/sapi.js
@@ -65,9 +65,10 @@ function SapiPoller(options) {
 
 	this.sp_agent = options.agent;
 
-	this.sp_sapi = new mod_restify.createJsonClient({
+	this.sp_sapi = mod_restify.createJsonClient({
 		url: this.sp_url,
 		agent: this.sp_agent,
+		log: this.sp_log,
 		requestTimeout: 10000,
 		retry: false
 	});
@@ -230,9 +231,12 @@ SapiPoller.prototype.state_runq_do = function (S) {
 	var self = this;
 	var vmapi = this.sp_vmapis[ent.dc];
 	if (vmapi === undefined) {
-		vmapi = new mod_restify.createJsonClient({
+		vmapi = mod_restify.createJsonClient({
 			url: 'http://vmapi.' + ent.dc + '.' + this.sp_suffix,
-			agent: this.sp_agent
+			agent: this.sp_agent,
+			log: this.sp_log,
+			requestTimeout: 10000,
+			retry: false
 		});
 		this.sp_vmapis[ent.dc] = vmapi;
 	}
